In [1]:
import pandas as pd

In [22]:
import pandas as pd
df = pd.read_csv('employee_data.csv', skiprows=1)
df

Employee_Name  EmpID  MarriedID  MaritalStatusID  GenderID  \
0         Adinolfi, Wilson  K  10026          0                0         1   
1    Ait Sidi, Karthikeyan     10084          1                1         1   
2           Akinkuolie, Sarah  10196          1                1         0   
3                Alagbe,Trina  10088          1                1         0   
4            Anderson, Carol   10069          0                2         0   
..                        ...    ...        ...              ...       ...   
306            Woodson, Jason  10135          0                0         1   
307        Ybarra, Catherine   10301          0                0         0   
308          Zamora, Jennifer  10010          0                0         0   
309               Zhou, Julia  10043          0                0         0   
310             Zima, Colleen  10271          0                4         0   

     EmpStatusID  DeptID  PerfScoreID  FromDiversityJobFairID  Salary  ...  \
0              1       5            4                       0   62506  ...   
1              5       3            3                       0  104437  ...   
2              5       5            3                       0   64955  ...   
3              1       5            3                       0   64991  ...   
4              5       5            3                       0   50825  ...   
..           ...     ...          ...                     ...     ...  ...   
306            1       5            3                       0   65893  ...   
307            5       5            1                       0   48513  ...   
308            1       3            4                       0  220450  ...   
309            1       3            3                       0   89292  ...   
310            1       5            3                       0   45046  ...   

        ManagerName  ManagerID  RecruitmentSource PerformanceScore  \
0    Michael Albert       22.0           LinkedIn          Exceeds   
1        Simon Roup        4.0             Indeed      Fully Meets   
2    Kissy Sullivan       20.0           LinkedIn      Fully Meets   
3      Elijiah Gray       16.0             Indeed      Fully Meets   
4    Webster Butler       39.0      Google Search      Fully Meets   
..              ...        ...                ...              ...   
306  Kissy Sullivan       20.0           LinkedIn      Fully Meets   
307  Brannon Miller       12.0      Google Search              PIP   
308      Janet King        2.0  Employee Referral          Exceeds   
309      Simon Roup        4.0  Employee Referral      Fully Meets   
310   David Stanley       14.0           LinkedIn      Fully Meets   

     EngagementSurvey EmpSatisfaction SpecialProjectsCount  \
0                4.60               5                    0   
1                4.96               3                    6   
2                3.02               3                    0   
3                4.84               5                    0   
4                5.00               4                    0   
..                ...             ...                  ...   
306              4.07               4                    0   
307              3.20               2                    0   
308              4.60               5                    6   
309              5.00               3                    5   
310              4.50               5                    0   

    LastPerformanceReview_Date DaysLateLast30 Absences  
0                    1/17/2019              0        1  
1                    2/24/2016              0       17  
2                    5/15/2012              0        3  
3                     1/3/2019              0       15  
4                     2/1/2016              0        2  
..                         ...            ...      ...  
306                  2/28/2019              0       13  
307                   9/2/2015              5        4  
308                  2/21/2019             

In [23]:
# See all column names
df.columns

Index(['Employee_Name', 'EmpID', 'MarriedID', 'MaritalStatusID', 'GenderID',
       'EmpStatusID', 'DeptID', 'PerfScoreID', 'FromDiversityJobFairID',
       'Salary', 'Termd', 'PositionID', 'Position', 'State', 'Zip', 'DOB',
       'Sex', 'MaritalDesc', 'CitizenDesc', 'HispanicLatino', 'RaceDesc',
       'DateofHire', 'DateofTermination', 'TermReason', 'EmploymentStatus',
       'Department', 'ManagerName', 'ManagerID', 'RecruitmentSource',
       'PerformanceScore', 'EngagementSurvey', 'EmpSatisfaction',
       'SpecialProjectsCount', 'LastPerformanceReview_Date', 'DaysLateLast30',
       'Absences'],
      dtype='object')

In [28]:
# How is the perfomance distribution of the employees in the company?
df['PerformanceScore'].value_counts()

PerformanceScore
Fully Meets          243
Exceeds               37
Needs Improvement     18
PIP                   13
Name: count, dtype: int64

**Business Interpretation:**
- 78% of employees "Fully Meet" expectations (243/311 employees)
- 12% "Exceed" expectations - this could indicate limited growth opportunities
- 10% are underperforming (18 + 13 = 31 employees needing improvement or on PIP)

**Assessment:** Overall performance distribution appears healthy for a 311-employee organization.

**Next Question:** Are performance issues evenly distributed across departments, or are certain departments struggling more than others?

In [50]:
# How is the perfomance distribution by department?
# Performance by department
df.groupby('Department')['PerformanceScore'].value_counts()

Department            PerformanceScore 
Admin Offices         Fully Meets            9
Executive Office      Fully Meets            1
IT/IS                 Fully Meets           42
                      Exceeds                6
                      Needs Improvement      1
                      PIP                    1
Production            Fully Meets          159
                      Exceeds               27
                      Needs Improvement     15
                      PIP                    8
Sales                 Fully Meets           24
                      PIP                    4
                      Exceeds                2
                      Needs Improvement      1
Software Engineering  Fully Meets            8
                      Exceeds                2
                      Needs Improvement      1
Name: count, dtype: int64

**Key Finding:** Production department shows disproportionate performance issues in raw numbers.

**The numbers:**
- Production: 23 underperformers (15 + 8)
- Sales: 5 underperformers (4 + 1)  
- IT/IS: 2 underperformers (1 + 1)

**Wait - Critical Analysis Question:** Are we comparing departments fairly? Production has 23 underperformers, but it also has way more employees than other departments. We need to calculate underperformance **rates**, not just raw counts.

**Next Analysis:** Let's calculate what percentage of each department is underperforming to get a true comparison.

In [53]:
# First, get total employees by department
dept_totals = df.groupby('Department').size()
print("Total employees by department:")
print(dept_totals)

# Calculate underperformance rates
underperformers = df[df['PerformanceScore'].isin(['PIP', 'Needs Improvement'])]
underperformer_counts = underperformers.groupby('Department').size()

# Calculate rates
underperformance_rates = (underperformer_counts / dept_totals * 100).round(1)
print("\nUnderperformance rates by department:")
print(underperformance_rates)

Total employees by department:
Department
Admin Offices             9
Executive Office          1
IT/IS                    50
Production              209
Sales                    31
Software Engineering     11
dtype: int64

Underperformance rates by department:
Department
Admin Offices            NaN
Executive Office         NaN
IT/IS                    4.0
Production              11.0
Sales                   16.1
Software Engineering     9.1
dtype: float64


**Critical Business Finding:** We have two distinct problems requiring different solutions.

**Sales Department Crisis:**
- 16.1% underperformance rate (1 in 6 employees)
- Small team (31 people) but direct revenue impact
- 5 underperformers potentially affecting company sales

**Production Department Challenge:**
- 11% underperformance rate (1 in 9 employees)  
- Large team (209 people) = 23 total underperformers
- Scale makes this a significant operational issue

**Strategic Implication:** Sales needs immediate intensive intervention due to high rate and revenue impact. Production needs systematic improvement due to scale.

**Next Questions:** 
1. Are these problems related to compensation?
2. Do they have different root causes?

**Hypothesis:** Are performance issues related to compensation disparities?

If poor performance correlates with low pay, we'd expect our problem departments (Sales and Production) to show compensation gaps.

**Analysis:** Let's examine average salaries by department.

In [31]:
# Average salary by department
df.groupby('Department')['Salary'].mean()

Department
Admin Offices            71791.888889
Executive Office        250000.000000
IT/IS                    97064.640000
Production               59953.545455
Sales                    69061.258065
Software Engineering     94989.454545
Name: Salary, dtype: float64

**Key Findings:**

- **Production:** $59,953 average (lowest paid department)
- **Sales:** $69,061 average (higher than Production but still below IT/Software)
- **IT/IS:** $97,064 average (62% more than Production)
- **Software Engineering:** $94,989 average (58% more than Production)

**The Story Emerges:**
- Production is significantly underpaid AND has 23 underperformers
- Sales is better paid than Production but still 29% below IT/Software levels
- High-performing departments (IT/Software) have much higher compensation

**Business Insight:** There appears to be a correlation between low pay and performance issues, but the relationship is complex.

**Next Question:** Are there other factors beyond pay affecting performance?

**Hypothesis Testing:** If low pay is the primary driver of performance issues, we'd expect to see low engagement and satisfaction in our problem departments.

**Analysis:** Let's examine employee engagement and satisfaction scores by department to see if they correlate with our performance and pay findings.

In [33]:
df.groupby('Department')['EngagementSurvey'].mean()

Department
Admin Offices           4.393333
Executive Office        4.830000
IT/IS                   4.154000
Production              4.129569
Sales                   3.818710
Software Engineering    4.061818
Name: EngagementSurvey, dtype: float64

In [34]:
df.groupby('Department')['EmpSatisfaction'].mean()

Department
Admin Offices           3.555556
Executive Office        3.000000
IT/IS                   3.960000
Production              3.861244
Sales                   4.032258
Software Engineering    4.090909
Name: EmpSatisfaction, dtype: float64

**Engagement & Satisfaction Findings:**
- Results show the employees with the lowest engagement and job satisfaction are in the Sales Department. Interesting to note that the Production department has high engagement metrics when compared to their salaries, meaning that the work performance is likely not being impacted by their salary but from other factors.
- This analysis shows a potential issue within the Sales department considering the Engagement Survey and Employee Satisfaction ratings. This requires a followup with more information, especially with the questions asked in the survey and an analysis of the problematic areas employees face, and how that reflected in the survey itself.
- In the Sales department I will further analyze management and work culture. I believe Sales is highly dependant on a good work culture, considering the volume of stressful work they face. This requires further analysis.

**Updated Business Insight:** Our analysis reveals two distinct problems requiring different solutions:

**Production Department:**
- Performance issues (11% underperformance) despite high engagement (4.13) and reasonable satisfaction (3.86)
- Suggests problems are operational/systemic rather than motivational
- Underpaid but engaged workforce indicates skills gaps, training needs, or process issues

**Sales Department:**
- Highest performance issues (16.1% underperformance) WITH lowest engagement (3.82)
- Classic signs of management/culture problems in high-pressure environment
- Better compensated than Production but still struggling with motivation

**Strategic Recommendation:** 
- **Production:** Focus on training, processes, and gradual pay equity
- **Sales:** Immediate management intervention and culture assessment required

**Next Steps:** Deep-dive into Sales management structure and work environment factors.

## Deep-Dive: Sales Management Analysis

**Business Question:** Are Sales performance issues concentrated under specific managers, indicating management effectiveness problems?

**Hypothesis:** If Sales has culture/management issues, we'd expect to see underperformers clustered under certain managers.

In [57]:
# Sales department only
sales_dept = df[df['Department'] == 'Sales']

# Performance by manager in Sales
sales_performance = sales_dept.groupby('ManagerName')['PerformanceScore'].value_counts()
print("Sales Performance by Manager:")
print(sales_performance)

# Manager team sizes in Sales
sales_manager_sizes = sales_dept.groupby('ManagerName').size()
print("\nSales Team Sizes by Manager:")
print(sales_manager_sizes)

Sales Performance by Manager:
ManagerName     PerformanceScore 
Debra Houlihan  Fully Meets           2
                Needs Improvement     1
Janet King      Fully Meets           1
John Smith      Fully Meets          12
                PIP                   2
Lynn Daneault   Fully Meets           9
                Exceeds               2
                PIP                   2
Name: count, dtype: int64

Sales Team Sizes by Manager:
ManagerName
Debra Houlihan     3
Janet King         1
John Smith        14
Lynn Daneault     13
dtype: int64


In [58]:
# Calculate underperformance rates by manager
sales_underperformers = sales_dept[sales_dept['PerformanceScore'].isin(['PIP', 'Needs Improvement'])]
sales_underperformer_counts = sales_underperformers.groupby('ManagerName').size()

# Calculate rates
sales_underperformance_rates = (sales_underperformer_counts / sales_manager_sizes * 100).round(1)
print("Sales Underperformance Rates by Manager:")
print(sales_underperformance_rates)

Sales Underperformance Rates by Manager:
ManagerName
Debra Houlihan    33.3
Janet King         NaN
John Smith        14.3
Lynn Daneault     15.4
dtype: float64


**Sales Management Analysis Results:**

**Underperformance rates by manager:**
- **Debra Houlihan:** 33.3% (1 of 3 employees) - Small team, high rate
- **John Smith:** 14.3% (2 of 14 employees) - Large team, moderate rate  
- **Lynn Daneault:** 15.4% (2 of 13 employees) - Large team, moderate rate
- **Janet King:** 0% (0 of 1 employee) - Single employee, performing well

**Critical Business Insight:** While Debra Houlihan shows a high rate, her team is only 3 people (small sample). The two main managers (John Smith and Lynn Daneault) show similar underperformance rates (~14-15%).

**Conclusion:** Sales performance issues are **NOT management-driven** but appear to be **systemic to the sales function itself.**

**Root Cause Assessment:** The evidence points to:
- High-stress nature of sales work affecting performance
- Possible skills/training gaps in sales methodology
- Market or territory challenges
- Inadequate support systems for sales teams

**Strategic Implication:** Focus on sales training, process improvement, and stress management rather than management changes.

## Strategic Recommendations for HR Director

### Executive Summary
Our analysis reveals two distinct workforce challenges requiring targeted interventions:
- **Production:** 11% underperformance due to systemic operational issues
- **Sales:** 16.1% underperformance due to role-specific stress and skills gaps

### Immediate Actions (30-60 days)

**Sales Department - Priority 1:**
1. **Conduct comprehensive sales training audit** - Identify specific skill gaps
2. **Implement stress management/wellness program** - Address low engagement (3.82)
3. **Review sales quotas and territory assignments** - Ensure realistic expectations
4. **Manager coaching program** - Support managers dealing with high-stress teams

**Production Department - Priority 2:**
1. **Skills assessment and targeted training** - Address performance gaps in engaged workforce
2. **Process improvement initiative** - Review workflows and inefficiencies
3. **Gradual pay equity adjustment** - Address $37K gap with IT/Software over 2 years

### Long-term Strategy (3-12 months)

**Department-Specific Solutions:**
- **Sales:** Restructure compensation model, implement CRM/sales tools
- **Production:** Invest in automation/technology, create clear advancement paths

**Company-Wide Initiatives:**
- Annual compensation benchmarking
- Manager effectiveness training program
- Enhanced employee engagement monitoring

### Success Metrics
- **Sales:** Reduce underperformance from 16.1% to <8% within 6 months
- **Production:** Improve performance while maintaining high engagement
- **Overall:** Achieve <5% company-wide underperformance rate

### Investment Required
- **Sales interventions:** $50K-75K (training, tools, programs)
- **Production improvements:** $150K-200K (training, process, gradual pay increases)
- **ROI:** Estimated 15-20% productivity improvement worth $500K+ annually